src/main/java/mate/academy/intro/service/BookServiceImpl.java

```diff
@@ -35,4 +36,25 @@ public BookDto getBookById(Long id) {
                 .orElseThrow(() -> new EntityNotFoundException("Book not found with id " + id));
         return bookMapper.bookToBookDto(book);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        if (bookRepository.existsById(id)) {
+            bookRepository.deleteById(id);
+        } else {
+            throw new EntityNotFoundException("Can not find book with id: " + id);
+        }
+    }
+
+    @Override
+    public BookDto updateBook(Long id, CreateBookRequestDto bookDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(()
+                        -> new EntityNotFoundException("Cant find book by id " + id));
+        book.setTitle(bookDto.getTitle());
+        book.setAuthor(bookDto.getAuthor());
+        book.setIsbn(bookDto.getIsbn());
+        book.setPrice(BigDecimal.valueOf(bookDto.getPrice()));
```

Comment:
Let’s use Mapper for this logic. Create method `Book updateBook(Book book, CreateBookRequestDto bookDto)`